File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/MicroserviceDeploymentStatus.java
package org.kinotic.management.api.model;

/**
 * Lifecycle state of a {@link MicroserviceDeployment}, and why when something is wrong.
 *
 * @param type    the lifecycle state of the deployment
 * @param message why the deployment is in this state, or null when the state speaks for
 *                itself; typically the failure that left it
 *                {@link MicroserviceDeploymentStatusType#FAILED}
 */
public record MicroserviceDeploymentStatus(MicroserviceDeploymentStatusType type, String message) {

    public MicroserviceDeploymentStatus(MicroserviceDeploymentStatusType type) {
        this(type, null);
    }
}

File: kinotic-js/workspace/packages/management-api/src/api/model/MicroserviceDeploymentStatus.ts
import { MicroserviceDeploymentStatusType } from '@/api/model/MicroserviceDeploymentStatusType'

/**
 * Lifecycle state of a MicroserviceDeployment, and why when something is wrong.
 */
export interface MicroserviceDeploymentStatus {
    /**
     * The lifecycle state of the deployment.
     */
    type: MicroserviceDeploymentStatusType
    /**
     * Why the deployment is in this state, or null when the state speaks for itself;
     * typically the failure that left it FAILED.
     */
    message: string | null
}
